### DISCUSSION TITLE
Imitation learning data collection expert query probability

### DISCUSSION THREAD

**Discussion by Sandbergo** [ds4dm/ecole#154] [Questions] [answered]
The sample generation for variable selection imitation learning models are mostly the same in Gasse et al. (2019) and the examples/branching-imitation.ipynb example, however I noticed that the probability of querying the strong branching rule was 0.60 in Gasse et al. (2019) and 0.05 in the example file. 

I would be very interested in hearing whether this is a deliberate change, and if so, what the reasoning behind this change is.

Best regards.

**Comment by gasse** [comment, marked as answer]
Dear Lars,

From what I see that probability is set to 0.05 in both case:
https://github.com/ds4dm/ecole/blob/master/examples/branching-imitation.ipynb
https://github.com/ds4dm/learn2branch/blob/master/02_generate_dataset.py#L322

A reason for setting the probability higher could be to obtain training samples faster, so we might have done that at some point for debugging. Could you point me to the file where you see this probability set to 0.60 ?

Best

**Reply by Sandbergo** [reply]
Hello,

My mistake, I was examining Prateek Guptas fork (https://github.com/pg2455/Hybrid-learn2branch/blob/master/02_generate_dataset.py), and did not notice that it was changed from the original. 

Thank you for your reply.

**Reply by gasse** [reply]
No worries, we're glad to take comments. And sometimes there are bugs in our code :P
Don't hesitate to report anything you find weird.

Best
